static/src/javascripts/projects/common/modules/commercial/create-ad-slot.js

```diff
@@ -102,49 +99,70 @@ define([
+                'data-name': name
+            })
+            .attr(attrs)
+            .addClass('js-ad-slot ad-slot ad-slot--dfp ' + classes.join(' '));
+    }
+
     return function (name, types, series, keywords, slotTarget) {
-        var attrName,
-            slotName = slotTarget ? slotTarget : name,
+        var slotName = slotTarget ? slotTarget : name,
+            attributes = {},
             definition,
-            dataAttrs,
+            classes,
             $adSlot;
 
-        definition = (slotName.match(/^inline/) && slotName !== 'inline1') ? adSlotDefinitions.inline : adSlotDefinitions[slotName];
+        definition = adSlotDefinitions[slotName] || adSlotDefinitions.inline;
+        name = definition.name || name;
+
         if (config.page.hasPageSkin && slotName === 'merchandising-high') {
             definition.sizeMappings.wide = '1,1';
         }
-        dataAttrs  = {
-            label:   definition.label !== undefined ? definition.label : true,
-            refresh: definition.refresh !== undefined ? definition.refresh : true
-        };
-        $adSlot = $.create(template(
-            adSlotTpl,
-            {
-                name: definition.name || name,
-                // badges now append their index to the name
-                normalisedName: (definition.name || name).replace(/((?:ad|fo|sp)badge).*/, '$1'),
-                types: types ? map((isArray(types) ? types : [types]), function (type) {
-                    return ' ad-slot--' + type;
-                }).join('') : '',
-                sizeMappings: map(pairs(definition.sizeMappings), function (size) {
-                    return ' data-' + size[0] + '="' + size[1] + '"';
-                }).join('')
-            })
-        );
-        for (attrName in dataAttrs) {
-            if (dataAttrs[attrName] === false) {
-                $adSlot.attr('data-' + attrName, 'false');
-            }
+
+        assign(attributes, definition.sizeMappings);
+
+        if (definition.label === false) {
+            attributes.label = 'false';
+        }
+
+        if (definition.refresh === false) {
+            attributes.refresh = 'false';
         }
+
         if (slotTarget) {
-            $adSlot.attr('data-slot-target', slotTarget);
+            attributes['slot-target'] = slotTarget;
         }
+
         if (series) {
-            $adSlot.attr('data-series', series);
+            attributes.series = series;
         }
+
         if (keywords) {
-            $adSlot.attr('data-keywords', keywords);
+            attributes.keywords = keywords;
+        }
+
+        if (types) {
+            classes = map((isArray(types) ? types : [types]), function (type) {
+                return 'ad-slot--' + type;
+            });
         }
+
+        classes.push('ad-slot--' + name.replace(/((?:ad|fo|sp)badge).*/, '$1'));
+
+        $adSlot = createAdSlotElement(
+            name,
+            transform(attributes, function (result, size, key) {
```

Comment:
This seems a tad esoteric to me. Why not have your attributes named `data-x` in the first instance? It's not really much more boilerplate but it's simpler to reason about, I think. I don't think I'd ever seen _.transform before.